src/core/MIGRATION.md

```diff
@@ -1056,15 +1056,14 @@ import { setup, start } from '../core_plugins/embeddables/public/legacy';
-| `import 'ui/apply_filters'`                            | `data.filter.loadLegacyDirectives`         | `loadLegacyDirectives()` should be called explicitly where you previously relied on importing for side effects                     |
-| `import 'ui/filter_bar'`                               | `data.filter.loadLegacyDirectives`         | `loadLegacyDirectives()` should be called explicitly where you previously relied on importing for side effects                     |
-| `import 'ui/query_bar'`                                | `data.query.loadLegacyDirectives`          | `loadLegacyDirectives()` should be called explicitly where you previously relied on importing for side effects                     |
-| `import 'ui/search_bar'`                               | `data.search.loadLegacyDirectives`         | `loadLegacyDirectives()` should be called explicitly where you previously relied on importing for side effects                     |
-| `import { QueryBar } from 'ui/query_bar'`              | `data.query.ui.QueryBar`                   | --                                                                                                                                 |
-| `import { SearchBar } from 'ui/search_bar'`            | `data.search.ui.SearchBar`                 | --                                                                                                                                 |
+| Legacy Platform                                        | New Platform                               | Notes                                                                                                                                  |
+|--------------------------------------------------------|--------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------|
+| `import 'ui/apply_filters'`                            | `import { ApplyFiltersPopover } from '../data/public'`         | `import '../data/public/legacy` should be called to load legacy directives                                               |
+| `import 'ui/filter_bar'`                               | `import { FilterBar } from '../data/public'`                   | `import '../data/public/legacy` should be called to load legacy directives                                               |
+| `import 'ui/query_bar'`                                | `import { QueryBar, QueryBarInput } from '../data/public'`     | Directives are deprecated.                                                                                            |
+| `import 'ui/search_bar'`                               | `import { SearchBar } from '../kibana_react/public'`           | Directive is deprecated.                                                                                              |
```

Comment:
I don't see `SearchBar`, `TopNavMenu` in `src/plugins/kibana_react` on master, I think we should hold off with merging until it's available.

I'm curios about the choice to put these react components there vs for instance `FilterBar` which is in the data plugin?

Reply:
* top_nav_menu is in the legacy plugin on `master`: `src/legacy/core_plugins/kibana_react/public/top_nav_menu`

* Generally speaking, we are trying to host only components that correspond with a specifc data service in the `data` plugin. `FilterManager` + `FilterBar` for example. 

* As @Bargs and @TinaHeiligers are building a search service ATM with their [Saved Queries PR](https://github.com/elastic/kibana/pull/39140), we just moved `SearchBar` back to `data` plugin. I updated the doc to reflect that.

Reply:
Discussed offline: I was getting confused about the relative locations and assumed these are NP plugins importing from other NP plugins. I suspect other people might also loose track of which API's (core or app-arch) need to be shimmed vs are already in the NP.

Maybe we can change this paragraph:
> In client code, we have a series of plugins which house shared application services that are being built in the shape of the new platform, but still technically reside in the legacy world. If your plugin depends on any of the APIs below, you'll need to add a dependency on the new platform plugin which will house them moving forward.

To something like:
> In client code, we have a series of plugins which house shared application services that are being built in the shape of the new platform, but for the time being, are only available in legacy. So if your plugin depends on any of the APIs below, you'll need build your plugin as a legacy plugin that shims the new platform. Once these API's have been moved to the new platform you can migrate your plugin and declare a dependency on the plugin that owns the API's you require.

Reply:
Updated. Thanks.

Mostly important, it seems like the implication of this, is that until `data` shim moves into new platform, anything depending on it - has to be a shim as well.

@lukeelmers